test/unit/BPool/BPool_Bind_Unbind.tree

```diff
@@ -0,0 +1,51 @@
+BPool::bind
+├── when reentrancy lock is NOT set
+│   ├── when called NOT by controller
+│   │   └── it should revert
+│   └── when called by controller
+│       ├── when pool is finalized
+│       │   └── it should revert
+│       ├── when token is already bound
+│       │   └── it should revert
+│       ├── when MAX_BOUND_TOKENS tokens are already bound
+│       │   └── it should revert
+│       ├── when token weight is too low
+│       │   └── it should revert
+│       ├── when token weight is too high
+│       │   └── it should revert
+│       ├── when too little balance is provided
+│       │   └── it should revert
+│       ├── when weight sum exceeds MAX_TOTAL_WEIGHT
+│       │   └── it should revert
+│       └── when it is NOT finalized
+│           ├── it sets reentrancy lock
+│           ├── it increments _totalWeight
+│           ├── it calls _pullUnderlying
+│           ├── it adds token to the list
+│           └── it sets the token record
+└── when reentrancy lock is set
+    └── it should revert
+
+BPool::unbind
+├── when reentrancy lock is NOT set
+│   ├── when called NOT by controller
+│   │   └── it should revert
+│   └── when called by controller
+│       ├── when token is not bound
+│       │   └── it should revert
+│       └── when token is bound
+│           ├── when pool is finalized
```

Comment:
i fear for the separations of this tree, didn't it bring a lot of modifiers?

Reply:
4 modifiers to be exact. I find it easier to see the preconditions of a particular test when they are on separate modifiers each of which is a separate word (and sometimes they end up in their own line!) instead of the precondition getting mangled into the test name. But will give it a try!

Reply:
Same, esp when linter puts them one per line, it give a clear overview of the precondition imo